Revamp Inference Providers doc (#1652)
huggingface/hub-docs: docs/api-inference/hub-integration.md

# Hub Integration

**TODO:**
- search for models (UI): https://huggingface.co/models?inference_provider=fireworks-ai&sort=trending
- search all models (UI): https://huggingface.co/models?inference_provider=all&sort=trending
- set custom API key from provider (optional)
- order providers by preference (apply to widget and code snippets)

### Features using Inference Providers

Several Hugging Face features utilize the Inference Providers API and count towards your monthly credits:

- Inference Widgets: Interactive widgets available on model pages.
- [Inference Playground](https://huggingface.co/playground): A comprehensive chat interface supporting various models and providers.

> - Data Studio AI: Converts text to SQL queries for datasets.

Reviewer: you can "delegate" to e.g. @mishig25, @cfahlgren1 etc to wrap a small blurb about those (with a screenshot)

Author: wouldn't mind some help :pray: 

For now I've taken screenshots from the website but can't say it looks so nice: https://moon-ci-docs.huggingface.co/docs/api-inference/pr_1652/en/hub-integration :grimacing:

# Hub Integration

> The Inference Providers is tightly integrated with the Hugging Face Hub. No matter which provider you use, the usage and billing will be centralized in your Hugging Face account.

Reviewer 2: ```suggestion
Inference Providers is tightly integrated with the Hugging Face Hub. No matter which provider you use, the usage and billing will be centralized in your Hugging Face account.
```

> The Inference Providers is tightly integrated with the Hugging Face Hub. No matter in which service you use it, the usage and billing will be centralized on your Hugging Face account.

Reviewer 3: ```suggestion
The Inference Providers is tightly integrated with the Hugging Face Hub. No matter which provider you use, the usage and billing will be centralized in your Hugging Face account.
```

## Model search

> When listing models on the Hub, you can filter to select models deployed on the inference provider for your choice. For example, to list all models deployed on Fireworks AI infra: https://huggingface.co/models?inference_provider=fireworks-ai.

Reviewer 3: ```suggestion
When listing models on the Hub, you can filter to select models deployed on the inference provider of your choice. For example, to list all models deployed on Fireworks AI infra: https://huggingface.co/models?inference_provider=fireworks-ai.
```

<div class="flex justify-center">
    <img class="block light:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/models-filter-by-provider-light.png"/>
    <img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/models-filter-by-provider-dark.png"/>
</div>

> It is also possible to select multiple providers or even all of them to filter all models that are available on at least 1 provider: https://huggingface.co/models?inference_provider=all.

Reviewer 3: ```suggestion
It is also possible to select all or multiple providers and filter their available models: https://huggingface.co/models?inference_provider=all.
```

<div class="flex justify-center">
    <img class="block light:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/models-filter-any-provider-light.png"/>
    <img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/models-filter-any-provider-dark.png"/>
</div>

## Features using Inference Providers

> Several Hugging Face features utilize the Inference Providers and count towards your monthly credits. The included monthly credits for PRO and Enterprise should cover moderate usage of these features for most users.

Reviewer 3: ```suggestion
Several Hugging Face features utilize Inference Providers and count towards your monthly credits. The included monthly credits for PRO and Enterprise should cover moderate usage of these features for most users.
```

</div>

- [Inference Playground](https://huggingface.co/playground): A comprehensive chat interface supporting various models and providers.

<div class="flex justify-center">
    <img class="block light:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/playground-example-light.png"/>
    <img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/playground-example-dark.png"/>
</div>

- [Data Studio AI](https://huggingface.co/datasets/open-r1/codeforces-cots/viewer): Converts text to SQL queries for datasets.

<div class="flex justify-center">
    <img class="block light:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/data-studio-example-light.png"/>
    <img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/data-studio-example-dark.png"/>
</div>

## User Settings

In your user account settings, you are able to:

> - set your own API keys for the providers you’ve signed up with. Otherwise, you can still use them – your requests will be billed on your HF account. More details in the [billing section](./pricing#routed-requests-vs-direct-calls).

Reviewer 2: ```suggestion
- set your own API keys for the providers you’ve signed up with. If you don't, your requests will be billed on your HF account. More details in the [billing section](./pricing#routed-requests-vs-direct-calls).
```
(shorter 🤷)
